### PR TITLE
Dtsbps 407 report controller

### PR DESCRIPTION
**Pull request by gauravchawla73463**
**Before creating a pull request make sure that:**

### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/DTSBPS-407

### Change description ###
new end point added

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/controllers/ReportsControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        );
+
+        mockMvc
+            .perform(get("/reports/count-summary?date=2021-03-28"))
+            .andExpect(status().isOk())
+            .andExpect(jsonPath("$.total_received").value(response.totalReceived))
+            .andExpect(jsonPath("$.total_rejected").value(response.totalRejected))
+            .andExpect(jsonPath("$.time_stamp").value(response.timeStamp.format(
+                DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss"))))
+            .andExpect(jsonPath("$.data.length()").value(2))
```
Comment (mdayican) [inline src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/controllers/ReportsControllerTest.java:70]: why the other data element did not asserted ?

File: src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/controllers/ReportsControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+            .andExpect(jsonPath("$.data[0].received").value(response.items.get(0).received))
+            .andExpect(jsonPath("$.data[0].rejected").value(response.items.get(0).rejected))
+            .andExpect(jsonPath("$.data[0].container").value(response.items.get(0).container))
+            .andExpect(jsonPath("$.data[0].date").value(response.items.get(0).date.toString()));
```
Comment (mdayican) [inline src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/controllers/ReportsControllerTest.java:74]: you should test response but you are asserting with response

File: src/main/java/uk/gov/hmcts/reform/blobrouter/data/reports/EnvelopeCountSummaryReportListResponse.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        int totalRejectedEnvelopes = 0;
+
+        for (var item : items) {
+            totalReceivedEnvelopes += item.received;
+            totalRejectedEnvelopes += item.rejected;
+        }
+
+        this.totalReceived = totalReceivedEnvelopes;
+        this.totalRejected = totalRejectedEnvelopes;
+        timeStamp = LocalDateTime.now().withNano(0);
```
Comment (mdayican) [inline src/main/java/uk/gov/hmcts/reform/blobrouter/data/reports/EnvelopeCountSummaryReportListResponse.java:39]: why `withNano` ?

File: src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/controllers/ReportsControllerTest.java (hunk 2 above)
Comment (apereverzin) [reply src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/controllers/ReportsControllerTest.java:74]: yes, the right way is to assert literal values

File: src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/controllers/ReportsControllerTest.java [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+            .andExpect(jsonPath("$.data[1].received").value(232))
+            .andExpect(jsonPath("$.data[1].rejected").value(19))
+            .andExpect(jsonPath("$.data[1].container").value(PCQ_CONTAINER))
+            .andReturn();
+        LocalDateTime endTime = LocalDateTime.now();
+        LocalDateTime jsonTimeStamp =
+            LocalDateTime.parse(
+                JsonPath.read(
+                    result.getResponse().getContentAsString(),
+                    "$.time_stamp"), DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss"));
```
Comment (mdayican) [inline src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/controllers/ReportsControllerTest.java:72]: formatting

**Review by mdayican — APPROVED** [review]

**Pull request merged**